internal/network/port_test.go

```diff
@@ -0,0 +1,30 @@
+package network
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsPortFree(t *testing.T) {
+	// Find a free port
+	port := 0
+	for port = 10000; port < 10100; port++ {
+		if IsPortFree(port) != nil {
+			break
+		}
+	}
+
+	// bind that port on localhost
+	l, err := bindAddress(LocalhostBindAddr(port))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer l.Close()
+
+	// assert that the port is now bound
+	err = IsPortFree(port)
```

Comment:
as you may have seen, this unit test passed on linux and failed on macos

According to [this](https://stackoverflow.com/a/14388707):
* macos allows simultaneously binding `127.0.0.1:1234` and `0.0.0.0:1234` if `SO_REUSEADDR` is set (and it turns out go does set that for listening sockets)
* linux doesn't

we could try turning off `SO_REUSEADDR` for our sockets?

Reply:
i just changed the interface to accept a bind address instead of a port, which is probably closer to what we want anyway.